odoo/orm/domains.py

```diff
@@ -984,11 +934,11 @@ def nary_optimization(optimization: Callable[[type[DomainNary], BaseModel, Optim
     then sorts them by (field, operator_type, operator) where operator type
     groups similar operators together.
     """
-    _CONDITION_MERGE_OPTIMIZATIONS.append(optimization)
+    _MERGE_OPTIMIZATIONS.append(optimization)
     return optimization
 
 
-def nary_condition_optimization(*, operators: Collection[str], field_condition: Callable[[Field], bool] | None = None):
+def nary_condition_optimization(operators: Collection[str], field_types: Collection[str] | None = None):
```

Comment:
Question: since the field_types is now a collection, how would I implement `in` merge? It applies to all field types that are not multi-valued, but they may be loaded after the domain module.

Reply:
Anyway the test on field types is done in the merge function called by `_optimize`.
So adding some `if` statement inside the merge function won't make a big difference, will it?
IMHO requiring a lambda inside a decorator is just higher-order masturbation, compared to a plain stupid `if` inside the function, unless it actually makes a difference in performance (not the case here).

Reply:
Ok